protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -255,4 +263,61 @@ public T parseBytes(byte[] serialized) {
       }
     }
   }
+
+  private static final class KnownLengthByteBufferIterable implements Iterable<ByteBuffer> {
+    private final InputStream stream;
+    private final int length;
+
+    private KnownLengthByteBufferIterable(InputStream stream, int length) {
+      this.stream = stream;
+      this.length = length;
+      stream.mark(length);
+    }
+
+    @Override
+    public Iterator<ByteBuffer> iterator() {
+      try {
+        stream.reset();
+        stream.mark(length);
+        return new Iterator<ByteBuffer>() {
+          private ByteBuffer buffer;
+
+          @Override
+          public boolean hasNext() {
+            if (buffer != null) {
+              return true;
+            }
+            try {
+              buffer = ((HasByteBuffer) stream).getByteBuffer(stream.available());
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+            return buffer != null;
+          }
+
+          @Override
+          public ByteBuffer next() {
+            if (!hasNext()) {
+              throw new NoSuchElementException();
+            }
+            ByteBuffer res = buffer;
+            try {
+              stream.skip(buffer.remaining());
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+            buffer = null;
+            return res;
+          }
+
+          @Override
+          public void remove() {
+            throw new UnsupportedOperationException();
+          }
+        };
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
It seems like this exception may not be caught.

Reply:
Obsolete.